cfme/infrastructure/host.py

```diff
@@ -329,6 +302,47 @@ def run_smartstate_analysis(self):
         sel.handle_alert()
         flash.assert_message_contain('"{}": Analysis successfully initiated'.format(self.name))
 
+    def check_compliance(self, timeout=240):
+        """Initiates compliance check and waits for it to finish."""
+        navigate_to(self, 'Details')
+        original_state = self.compliance_status
+        tb.select('Policy', 'Check Compliance of Last Known Configuration', invokes_alert=True)
+        sel.handle_alert()
+        flash.assert_no_errors()
+        wait_for(
+            lambda: self.compliance_status != original_state,
+            num_sec=timeout, delay=5, message="compliance of {} checked".format(self.name)
+        )
+        return self.compliant
```

Comment:
I don't think this should return current compliance status. It should only ensure that the appropriate button is clicked and wait for the compliance check to finish. If we want to assure that the host is compliant after running this, we should do `assert host.is_compliant`.